apps/dashboard/src/components/workflow-editor/workflow-canvas.tsx

```diff
@@ -51,40 +51,42 @@ const panOnDrag = [1, 2];
 const Y_DISTANCE = NODE_HEIGHT + 50;
 
 const mapStepToNode = (
-  step: StepResponseDto,
-  previousPosition: { x: number; y: number }
+  step: StepDto,
+  previousPosition: { x: number; y: number },
+  addStepIndex: number
 ): Node<NodeData, keyof typeof nodeTypes> => {
   let content = '';
   if (step.type === StepTypeEnum.DELAY) {
     content = `Wait to send ~ 30 minutes`;
   }
 
   return {
-    id: step.stepUuid,
+    id: crypto.randomUUID(),
     position: { x: previousPosition.x, y: previousPosition.y + Y_DISTANCE },
     data: {
       name: step.name,
       content,
+      addStepIndex,
```

Comment:
the index information is used to know where the step should be added in the array of steps